render.go

```diff
@@ -0,0 +1,87 @@
+package render
+
+import (
+	"image/color"
+	"image/draw"
+	"math"
+
+	"github.com/benoitkugler/textlayout/fonts"
+	"github.com/go-text/typesetting/shaping"
+	"github.com/srwiley/rasterx"
+	"golang.org/x/image/math/fixed"
+)
+
+type Renderer struct {
+	FontSize, PixScale float32
+	Color              color.Color
+}
+
+func (r *Renderer) DrawString(str string, img draw.Image, face fonts.Face) int {
+	sh := &shaping.HarfbuzzShaper{}
+	in := shaping.Input{
+		Text:     []rune(str),
+		RunStart: 0,
+		RunEnd:   len(str),
+		Face:     face,
+		Size:     fixed.I(int(r.FontSize * r.PixScale)),
+	}
+	out := sh.Shape(in)
+	return r.DrawShapedRunAt(out, img, 0, out.LineBounds.Ascent.Ceil())
+}
+
+func (r *Renderer) DrawStringAt(str string, img draw.Image, x, y int, face fonts.Face) int {
+	sh := &shaping.HarfbuzzShaper{}
+	in := shaping.Input{
+		Text:     []rune(str),
+		RunStart: 0,
+		RunEnd:   len(str),
+		Face:     face,
+		Size:     fixed.I(int(r.FontSize * r.PixScale)),
+	}
+	return r.DrawShapedRunAt(sh.Shape(in), img, x, y)
+}
```

Comment:
Should these two methods wrap the string to fit within the provided image (as best as possible, anyway)?

Reply:
That is a good question. I'm not sure. As we are not passing any sizing info it may overflow on the X or the Y.
From a "canvas" point of view my assumption had been that basic text paint isn't exposing smarts and you would use the run painting variant if you want to do anything clever like wrap or multi-font etc.
As I guess the API wouldn't change perhaps we land this simplistic version and then if it is expected we can wrap later?

Reply:
Sure!